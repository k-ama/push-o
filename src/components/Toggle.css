.Toogle{
  position: relative;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}
.Toggle .Toggle-label{
  width: 100%;
  height: 100%;
  border: 0;
  transition: background-color 2s;
}
