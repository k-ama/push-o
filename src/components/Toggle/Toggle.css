.Toggle {
  height: 2rem;
  width: 3rem;
  display: flex;
  align-items: center;
  padding: 0 0.5rem;
  background-color:red;
  transition: background-color 0.25s ease-in-out;
}
.Toggle.Toggle-on {
  background-color: grey;
}
.Toggle .Toggle-indicator {
  height: 1.5rem;
  width: 1.5rem;
  background-color: white;
  transition: transform 0.25s ease-in-out;
}
.Toggle.Toggle-on .Toggle-indicator {
  transform: translateX(1.5rem);
}
