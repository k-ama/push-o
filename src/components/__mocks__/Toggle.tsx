import React from "react"
import { ReactElement } from "react"

const ToggleMock = (): ReactElement => {
  return <div className="Toggle"/>
}

export default ToggleMock