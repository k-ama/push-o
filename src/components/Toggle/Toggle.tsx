import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { Actions } from '../../store/actions'
import './Toggle.css';

export const Toggle = () => {

  return (
    <div className="Toggle">
      <label className="Toggle-label">
        <input type="checkbox"/>
        <span></span>
      </label>
    </div>
  );
}

export default Toggle;
