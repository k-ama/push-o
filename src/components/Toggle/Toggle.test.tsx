import React from 'react'
import Toggle from './Toggle'
import renderer from 'react-test-renderer'

describe('Toggle', () => {

  describe('Snapshot', () => {
    test('renders Toggle', () => {
      const component = renderer.create(<Toggle/>)
      expect(component.toJSON()).toMatchSnapshot()
    })
  })

})
