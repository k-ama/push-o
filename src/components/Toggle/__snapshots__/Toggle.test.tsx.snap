// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Snapshot renders Toggle 1`] = `
<div
  className="Toggle Toggle-on"
>
  <div
    className="Toggle-indicator"
    onClick={[Function]}
  />
</div>
`;
